### PR TITLE
[v1-007 & v1-009] Added New SignIn page & Responsive Design

### PR DESCRIPTION
**Pull request by NikitaKhadnevich**
This pull includes next changes:
- Added new folders for components;
- Added file-components as like Sign Form( contain sub-components for this Form) , FormikWrapper for this form, MainGrid component etc.
- Created Fokmik and Yup constants and rules for work;
- Deploy other constants and implement in project
- Added styles for components;
- Added basic React Router for work and testing link for this. 
As result we have working form, which can receive User data and work on  requires

### CODE REVIEW

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -1,7 +1,22 @@
...
+import { BrowserRouter as Router, Route, Routes, Link } from 'react-router-dom';
+import { SignInContainer, signIn } from './AppReceiver';
 
 function App() {
-  return <></>;
+  return (
+    <>
+      <CssBaseline />
+      <Router>
+        <Link to={signIn} style={{ width: '100%', textAlign: 'center', textDecoration: 'none' }}>
```
Comment (hauteballe) [inline frontend/src/App.tsx:11]: Please move all the styles inside separate file

File: frontend/src/AppReceiver.tsx [hunk 2]
```diff
@@ -0,0 +1,4 @@
+import SignInContainer from './pages/SignPages/SignInContainer';
+import { signIn, signUp, myCourses, myProfile, requests } from './config/constants/navigationPath';
```
Comment (hauteballe) [inline frontend/src/AppReceiver.tsx:2]: Please use absolute paths here

File: frontend/src/components/SignForms/SignInForm/FormikReceiver.tsx [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import signInSchema from "../../../utils/signHelpers/FormikSchemas/signIN_Schema";
+import initSignInvalue from "../../../config/constants/formikConstants";
+import { authLabel,  signUpLabel, afterSignUp } from '../../../config/constants/signConstants';
```
Comment (hauteballe) [inline frontend/src/components/SignForms/SignInForm/FormikReceiver.tsx:3]: Absolute paths too

File: frontend/src/components/SignForms/SignInForm/FormikInContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import SignIn from './SignIn';
+
+import { signInSchema, initSignInvalue } from './FormikReceiver';
+
+const FormikIn = (): any => {
+  const formik = useFormik({
+    initialValues: initSignInvalue,
+    validationSchema: signInSchema,
+    onSubmit: (values, { resetForm }): void => {
+      console.log(values);
```
Comment (hauteballe) [inline frontend/src/components/SignForms/SignInForm/FormikInContainer.tsx:12]: console.log

File: frontend/src/utils/signHelpers/FormikSchemas/signIN_Schema.tsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { string, object } from 'yup';
+import {
+  login,
+  signInMinHeight,
+  signInMaxHeight,
+  requiredLogin,
+  requiredPass,
+} from '../signReceiver';
+
+const signInSchema = object().shape({
```
Comment (hauteballe) [inline frontend/src/utils/signHelpers/FormikSchemas/signIN_Schema.tsx:10]: Please move all the validations to the separate folder src/validations (signInValidationSchema)

File: frontend/src/utils/signHelpers/signReceiver.tsx [hunk 6]
```diff
@@ -0,0 +1,3 @@
+import { login, signInMinHeight, signInMaxHeight, requiredLogin, requiredPass } from "../../config/constants/signConstants";
```
Comment (hauteballe) [inline frontend/src/utils/signHelpers/signReceiver.tsx:1]: Absolute path

File: frontend/src/components/SignForms/SignInForm/SignComponents/EmailField.tsx [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { TextField } from '@mui/material';
+import { EmailProps } from '../../SignInterfaces';
+import InputStyled from './styled';
+
+const EmailField = ({ login, touched, onChangeAccum, handleBlur, errors }: EmailProps) => (
+  <InputStyled
```
Comment (hauteballe) [inline frontend/src/components/SignForms/SignInForm/SignComponents/EmailField.tsx:7]: I think here should be basic custom input from src/components later ( but it's not merged to develop yet )

File: frontend/src/components/SignForms/SignInForm/SignComponents/PasswordField.tsx [hunk 8]
```diff
@@ -0,0 +1,24 @@
+/* eslint-disable react/jsx-no-bind */
+import React from 'react';
+import TextField from '@mui/material/TextField';
+import { PasswordProps } from '../../SignInterfaces';
+import InputStyled from './styled';
```
Comment (hauteballe) [inline frontend/src/components/SignForms/SignInForm/SignComponents/PasswordField.tsx:5]: Please check all your imports and divide them into 3 groups: libs, absolute paths and relative paths (there must be an empty line between them)

File: frontend/src/components/SignForms/SignPresenter/SignPresenter.tsx [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+          }}>
+          <Typography variant="h3" style={{marginRight: '1rem', fontWeight: 600}}>
+          {techArtLabel} 
+          </Typography>
+          <Typography variant="h3" >
+            {techLogIn}
+          </Typography>
+        </Box>
+        <Box style={{marginTop: '20px'}}>
+          <Typography variant="h5" style={{color: 'gray'}}>
```
Comment (hauteballe) [inline frontend/src/components/SignForms/SignPresenter/SignPresenter.tsx:25]: styles

File: frontend/src/pages/SignPages/SignReceiver.tsx [hunk 10]
```diff
@@ -0,0 +1,5 @@
+import FormikInContainer from '../../components/SignForms/SignInForm/FormikInContainer';
```
Comment (hauteballe) [inline frontend/src/pages/SignPages/SignReceiver.tsx:1]: absolute paths

**Review by hauteballe — CHANGES_REQUESTED** [review]
Some changes required

File: frontend/src/App.tsx (hunk 1 above)
Comment (NikitaKhadnevich) [reply frontend/src/App.tsx:11]: Its only test link for demo.  In 003 we will replace this Link components.

File: frontend/src/AppReceiver.tsx (hunk 2 above)
Comment (NikitaKhadnevich) [reply frontend/src/AppReceiver.tsx:2]: Added

File: frontend/src/components/SignForms/SignInForm/FormikReceiver.tsx (hunk 3 above)
Comment (NikitaKhadnevich) [reply frontend/src/components/SignForms/SignInForm/FormikReceiver.tsx:3]: Fixed

File: frontend/src/components/SignForms/SignInForm/FormikInContainer.tsx (hunk 4 above)
Comment (NikitaKhadnevich) [reply frontend/src/components/SignForms/SignInForm/FormikInContainer.tsx:12]: Removed

File: frontend/src/utils/signHelpers/FormikSchemas/signIN_Schema.tsx (hunk 5 above)
Comment (NikitaKhadnevich) [reply frontend/src/utils/signHelpers/FormikSchemas/signIN_Schema.tsx:10]: Replaced in scr and renamed

File: frontend/src/utils/signHelpers/signReceiver.tsx (hunk 6 above)
Comment (NikitaKhadnevich) [reply frontend/src/utils/signHelpers/signReceiver.tsx:1]: Fixed

File: frontend/src/components/SignForms/SignInForm/SignComponents/EmailField.tsx (hunk 7 above)
Comment (NikitaKhadnevich) [reply frontend/src/components/SignForms/SignInForm/SignComponents/EmailField.tsx:7]: Let's leave it as it is. When 003 will drop I will replace it or fix it otherwise

File: frontend/src/components/SignForms/SignInForm/SignComponents/PasswordField.tsx (hunk 8 above)
Comment (NikitaKhadnevich) [reply frontend/src/components/SignForms/SignInForm/SignComponents/PasswordField.tsx:5]: Fixed

File: frontend/src/components/SignForms/SignPresenter/SignPresenter.tsx (hunk 9 above)
Comment (NikitaKhadnevich) [reply frontend/src/components/SignForms/SignPresenter/SignPresenter.tsx:25]: Replaced in styled.tsx

File: frontend/src/pages/SignPages/SignReceiver.tsx (hunk 10 above)
Comment (NikitaKhadnevich) [reply frontend/src/pages/SignPages/SignReceiver.tsx]: Fixed

File: frontend/src/App.tsx (hunk 1 above)
Comment (k-krasilnikova) [reply frontend/src/App.tsx:11]: But could you please move all the styles of it to the styled file?

File: frontend/src/AppReceiver.tsx [hunk 11]
```diff
@@ -0,0 +1,4 @@
+import SignInContainer from './pages/SignPages/SignInContainer';
```
Comment (k-krasilnikova) [inline frontend/src/AppReceiver.tsx:1]: Do we really need this file? Let's remove it

File: frontend/src/components/SignForms/SignInForm/FormikInContainer.tsx [hunk 12]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
```
Comment (k-krasilnikova) [inline frontend/src/components/SignForms/SignInForm/FormikInContainer.tsx:1]: Remove please this file and move all the logic inside SignIn container
Rename please all the path to the: `frontend/src/pages/SignIn/SignInContainer.tsx` and `frontend/src/pages/SignIn/SignIn.tsx`

File: frontend/src/components/SignForms/SignInForm/SignComponents/SignLabel.tsx [hunk 13]
```diff
@@ -0,0 +1,13 @@
+/* eslint-disable react/jsx-no-bind */
+import React from 'react';
+
+import { Typography } from '@mui/material';
+import { authLabel } from '../FormikReceiver';
+
+const SignLabel = (): any => (
```
Comment (k-krasilnikova) [inline frontend/src/components/SignForms/SignInForm/SignComponents/SignLabel.tsx:7]: Please don't use `any` type anywhere, it's not safe enough

Comment (k-krasilnikova) [inline frontend/src/components/SignForms/SignInForm/SignComponents/SignLabel.tsx:7]: I'm not sure that it should be a separated component

File: frontend/src/components/SignForms/SignInForm/SignIn.tsx [hunk 14]
```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+import Button from '@material-ui/core/Button';
+
+import EmailField from './SignComponents/EmailField';
+import PasswordField from './SignComponents/PasswordField';
+import SignLabel from './SignComponents/SignLabel';
+import { SignInFormik } from '../SignInterfaces';
```
Comment (k-krasilnikova) [inline frontend/src/components/SignForms/SignInForm/SignIn.tsx:7]: Remove this file and change the structure of SignIn page

File: frontend/src/components/SignForms/SignInForm/SignIn.tsx [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+    errors,
+    touched,
+    handleChange,
+    isValid,
+    setFieldTouched,
+    handleBlur,
+    handleSubmit,
+  } = formik;
+
+  const onChangeAccum = (name: boolean, e: string): any => {
```
Comment (k-krasilnikova) [inline frontend/src/components/SignForms/SignInForm/SignIn.tsx:23]: any?
What is Accum?
What is true and false sequences? Do we really need this method?

File: frontend/src/components/SignForms/SignInterfaces.tsx [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+    },
+    handleChange: (e: any) => any,
+    isValid: boolean,
+    setFieldTouched: (name: any, value: any, isValidate?: any) => any,
+    handleBlur: (e: any) => any,
+    handleSubmit: (e: any) => any
+    values: { 
+      login: string,
+      password: string | number
+    },
```
Comment (k-krasilnikova) [inline frontend/src/components/SignForms/SignInterfaces.tsx:19]: Remove all the `any` types

File: frontend/src/components/SignForms/SignPresenter/SignPresenter.tsx [hunk 17]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { Box } from '@mui/material';
+
+import { authGreetings, techLogIn, techArtLabel } from './SignPresReceiver';
+import { SignPresenterBox, PresenterWrapper, SighTechArtLabel, SighPresentSubtext } from './styled';
+
+const SignPresenter = (): any => {
+  return (
```
Comment (k-krasilnikova) [inline frontend/src/components/SignForms/SignPresenter/SignPresenter.tsx:8]: any?
There is no need to use `return` statement

File: frontend/src/config/constants/formikConstants.tsx [hunk 18]
```diff
@@ -0,0 +1,13 @@
+interface FormikInit {
```
Comment (k-krasilnikova) [inline frontend/src/config/constants/formikConstants.tsx:1]: wht if i ll wrt cmmnts th sm wy? Rename to something more clear please

File: frontend/src/config/constants/navigationPath.tsx [hunk 19]
```diff
@@ -0,0 +1,10 @@
+const path = {
+  signIn: 'signin',
+  signUp: 'signup',
+  myCourses: 'mycourses',
+  myProfile: 'profile',
+  requests: 'requests',
+};
+
+const { signIn, signUp, myCourses, myProfile, requests } = path;
+export { signIn, signUp, myCourses, myProfile, requests };
```
Comment (k-krasilnikova) [inline frontend/src/config/constants/navigationPath.tsx:10]: What is it? We already have some routing in v1-003 brunch. Reuse them please

File: frontend/src/pages/SignPages/SignInContainer.tsx [hunk 20]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+import { FormikInContainer, SignPresenter } from './SignReceiver';
+
+import { SignMain, SignMainGrid, SignPresGrid, SignFormGrid } from './styled';
+
+const SignInContainer = (): any => {
+  return (
+    <SignMain>
```
Comment (k-krasilnikova) [inline frontend/src/pages/SignPages/SignInContainer.tsx:9]: any? return?

File: frontend/src/pages/SignPages/SignInContainer.tsx [hunk 21]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+import { FormikInContainer, SignPresenter } from './SignReceiver';
+
```
Comment (k-krasilnikova) [inline frontend/src/pages/SignPages/SignInContainer.tsx:4]: Unexpected empty line

File: frontend/src/themeSettings.ts [hunk 22]
```diff
@@ -0,0 +1,30 @@
+import { createTheme } from '@mui/material';
+
+const colors = {
```
Comment (k-krasilnikova) [inline frontend/src/themeSettings.ts:3]: uppercase?

**Review by k-krasilnikova — CHANGES_REQUESTED** [review]
Great job, but still some changes needed

File: frontend/src/config/constants/navigationPath.tsx (hunk 19 above)
Comment (NikitaKhadnevich) [reply frontend/src/config/constants/navigationPath.tsx:10]: I had created it before merge v003 into develop branch

File: frontend/src/components/SignForms/SignInForm/FormikInContainer.tsx (hunk 12 above)
Comment (NikitaKhadnevich) [reply frontend/src/components/SignForms/SignInForm/FormikInContainer.tsx:1]: _**Remove please this file and move all the logic inside SignIn container**_ 
Lets see, in future I will use useQuery and i want to add this into FormikContainer - now SignIn already include over 70n lines, and its almost all only Sign-form describe. If i will add logic yet it will be not fine, this file wll include over then 120-150 lines. Maybe i will rename FormikContainer to SignContainer and

**_Rename please all the path to the: frontend/src/pages/SignIn/SignInContainer.tsx and frontend/src/pages/SignIn/SignIn.tsx_**
will rename it as SignPage.tsx and also will fix path.

Comment (hauteballe) [reply frontend/src/components/SignForms/SignInForm/FormikInContainer.tsx:1]: You should have SignIn folder with SignIn.tsx and SignInContainer.tsx inside it. Now we have too much extra folders and components here

File: frontend/src/App.tsx (hunk 1 above)
Comment (NikitaKhadnevich) [reply frontend/src/App.tsx:11]: Yup, Sure

File: frontend/src/components/SignForms/SignInForm/FormikInContainer.tsx (hunk 12 above)
Comment (NikitaKhadnevich) [reply frontend/src/components/SignForms/SignInForm/FormikInContainer.tsx:1]: Ok, I see that folders-hell))) replaced and fixed paths and remove some files

File: frontend/src/components/SignForms/SignInForm/SignComponents/SignLabel.tsx (hunk 13 above)
Comment (NikitaKhadnevich) [reply frontend/src/components/SignForms/SignInForm/SignComponents/SignLabel.tsx:7]: Already fixed

File: frontend/src/components/SignForms/SignInForm/SignIn.tsx (hunk 14 above)
Comment (NikitaKhadnevich) [reply frontend/src/components/SignForms/SignInForm/SignIn.tsx:7]: Done

File: frontend/src/components/SignForms/SignInForm/SignIn.tsx (hunk 15 above)
Comment (NikitaKhadnevich) [reply frontend/src/components/SignForms/SignInForm/SignIn.tsx:23]: renamed onChangeAccum => onChangeHandler.

File: backend/package.json [hunk 23]
```diff
@@ -32,6 +32,7 @@
         "body-parser": "^1.19.1",
         "dotenv": "^10.0.0",
         "express": "^4.17.1",
-        "jest": "^27.4.5"
+        "jest": "^27.4.5",
+        "tsconfig-paths": "^3.12.0"
```
Comment (hauteballe) [inline backend/package.json:36]: Why did you use "tsconfig-paths" lib? Absolute paths are already working with 'baseUrl' setting
Please remove it

File: backend/tsconfig.json [hunk 24]
```diff
@@ -8,6 +8,17 @@
...
+      "paths": {
+        "src/*": [
+          "./src/*"
+        ]
+      }     
+    },
+    "include": ["./src/**/*.ts"],
+    "exclude": ["node_modules", "dist"],
+    "ts-node": {
+      "require": ["tsconfig-paths/register"]
```
Comment (hauteballe) [inline backend/tsconfig.json:22]: What is the purpose? Please change it according to the tsconfig.json on develop branch

File: frontend/.env [hunk 25]
```diff
@@ -1,5 +0,0 @@
-REACT_APP_API_URL=/api
-
-BROWSER=none
-PORT=3000
-PUBLIC_URL=/
```
Comment (hauteballe) [inline frontend/.env:5]: Why did you remove this part? Have you discussed these changes with someone?

File: frontend/src/components/SignForms/SignInForm/FormikInContainer.tsx (hunk 12 above)
Comment (hauteballe) [reply frontend/src/components/SignForms/SignInForm/FormikInContainer.tsx:1]: SignInForm is not a reusable component, so it rather should be in pages/SignIn folder

File: frontend/src/components/SignForms/SignInForm/styled.tsx [hunk 26]
```diff
@@ -0,0 +1,42 @@
...
+  maxWidth: '350px',
+  boxShadow: '0px 0px 10px 5px gray',
+});
+
+const FormBox = styled(Box)({
+  margin: '30px',
+});
+
+const ItemsBox = styled(Box)({
+  margin: '0px !important',
```
Comment (hauteballe) [inline frontend/src/components/SignForms/SignInForm/styled.tsx:17]: Do we really need to use important directive in all styled.ts files?

File: frontend/src/pages/SignPages/SignReceiver.tsx [hunk 27]
```diff
@@ -0,0 +1,5 @@
+import FormikInContainer from 'components/SignForms/SignInForm/FormikInContainer';
+import SignPresenter from 'components/SignForms/SignPresenter/SignPresenter';
```
Comment (hauteballe) [inline frontend/src/pages/SignPages/SignReceiver.tsx:2]: Why do you use SignPages and SignForms naming if there is only one page and form?

File: frontend/src/components/SignForms/SignInForm/SignIn.tsx (hunk 15 above)
Comment (NikitaKhadnevich) [reply frontend/src/components/SignForms/SignInForm/SignIn.tsx:23]: @k-krasilnikova renamed "onChangeAccum => onChangeHandler", any replaced. On my opinion its elegant approach for work with input data and errors if input data wrong. Lets see, Usually when the User starting enter the data called onChange and after we will handle result and show in additional filed under input error info or other info. "_<input ..../> + isError <p>Error</p>_". But this meta work otherwise - we will bind onChangeHandler and after every User tap will call onChangeHandler after/im call onChange and finally calling setFieldTouched. This method has 3 arg - name, true, false. **Name** - its input name for identificate. **True** - its isTouched status(if user starting enter the data isTouched anyway will be true) and last arg - **False** its helper which explicitly prevent/skip validation for every tap. Magic in "helperText" prop - helperText={touched.fieldname && errors.fieldname). each keyboard click in input will call onChangeHandler and based on input data and will observe - if smth wrong in touched(touched will be is not empty)  immediately helperText show to User wrong if all right - touched will be empty. We dont need to create additional fields for work as like paragraph and. also there are not problem with fields and code will be clear - this method fill only 3 lines. fuuh)) Lets save it?)

**Review by hauteballe — CHANGES_REQUESTED** [review]
And please fix all linter issues
screen

File: frontend/src/components/SignForms/SignInterfaces.tsx (hunk 16 above)
Comment (NikitaKhadnevich) [reply frontend/src/components/SignForms/SignInterfaces.tsx:19]: Removed wherever possible.

File: frontend/src/components/SignForms/SignPresenter/SignPresenter.tsx (hunk 17 above)
Comment (NikitaKhadnevich) [reply frontend/src/components/SignForms/SignPresenter/SignPresenter.tsx:8]: Fixed

File: frontend/src/config/constants/formikConstants.tsx (hunk 18 above)
Comment (NikitaKhadnevich) [reply frontend/src/config/constants/formikConstants.tsx:1]: Renamed

File: frontend/src/config/constants/navigationPath.tsx (hunk 19 above)
Comment (NikitaKhadnevich) [reply frontend/src/config/constants/navigationPath.tsx:10]: Fixed it add - signIn into this file

File: frontend/src/pages/SignPages/SignInContainer.tsx (hunk 20 above)
Comment (NikitaKhadnevich) [reply frontend/src/pages/SignPages/SignInContainer.tsx:9]: Fixed

File: frontend/src/themeSettings.ts (hunk 22 above)
Comment (NikitaKhadnevich) [reply frontend/src/themeSettings.ts:3]: Fixed

File: backend/package.json (hunk 23 above)
Comment (NikitaKhadnevich) [reply backend/package.json:36]: I dont fix this file - its merge from develop branch before

File: backend/tsconfig.json (hunk 24 above)
Comment (NikitaKhadnevich) [reply backend/tsconfig.json:22]: I dont fix this file - its merge from develop branch before

File: frontend/.env (hunk 25 above)
Comment (NikitaKhadnevich) [reply frontend/.env:5]: oh yes( I tested SPA and forgot return this settings.I will correct its immediately.

File: frontend/src/components/SignForms/SignInForm/styled.tsx (hunk 26 above)
Comment (NikitaKhadnevich) [reply frontend/src/components/SignForms/SignInForm/styled.tsx:17]: yup, because basic material ui styles covering margins/paddings and some styles. Without this in future there may be errors in deploy.

File: frontend/src/pages/SignPages/SignReceiver.tsx (hunk 27 above)
Comment (NikitaKhadnevich) [reply frontend/src/pages/SignPages/SignReceiver.tsx:2]: I renamed/deleted it

File: frontend/src/AppReceiver.tsx (hunk 11 above)
Comment (NikitaKhadnevich) [reply frontend/src/AppReceiver.tsx:1]: I replace it and added other files. Maybe will this file will be receiver for import? For example, path, components, other constants

File: frontend/src/pages/SignPages/SignInContainer.tsx (hunk 21 above)
Comment (NikitaKhadnevich) [reply frontend/src/pages/SignPages/SignInContainer.tsx:4]: Fixed

**Comment by NikitaKhadnevich** [comment]
This commit also include **`v1-009 feature`**. For this i had created:
1. Custom resolutions as like in conditions (`320px, 425px, 768px, 1024px, 1440px+`);
2. Added and fix basic MIU styles in `styled.tsx` files;
3. Created `@media` for responsible design in styled files
4. Tested and implement its^^^.
If `v1-009 feature` have to have separate pull/commit, i will overwrite its. Maybe we will have new SPA design in future.

**Comment by NikitaKhadnevich** [comment]
_**This pull request includes:**_
1. Refactored lines and folders;
2. Rewrote, renamed and united / split some components;
3. Added more clear methods and functions in containers;
4. Created new and replaced old TS types
5. "Drew" Styles for Responsive design and new resolutions
6. other fix
